src/main/java/service/DataConverterImpl.java

```diff
@@ -0,0 +1,22 @@
+package service;
+
+import core.basesyntax.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            String[] parts = line.split(",");
+            FruitTransaction.Operation operation
+                    = FruitTransaction.Operation.valueOf(parts.toString());
+            String fruit = Arrays.toString(parts);
+            int quantity = Integer.parseInt(Arrays.toString(parts));
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
+        }
+        return transactions;
+    }
```

Comment:
This method does not handle the possibility of `NumberFormatException` or `IllegalArgumentException` which may occur if the input data is not in the expected format. It would be safer to add a try-catch block around the parsing logic to handle these exceptions gracefully.